### DISCUSSION TITLE
gtk: extremely pixelated titlebar fonts on opensuse

### DISCUSSION THREAD

**Discussion by x0ba** [ghostty-org/ghostty#2792] [Q&A]
As seen in below photo, the font rendered in the titlebar and context menus in Ghostty is very pixelated. This only happens on my openSUSE laptop, not my Arch desktop. All other libadwaita apps look perfectly fine.

![Image](https://github.com/user-attachments/assets/5f7f3c97-84f5-49ac-815d-cb21599d76ac)

Steps to reproduce:
1. Install the `gtk4-devel` (version 4.16.5) and `libadwaita-devel` (version 1.6.1) on openSUSE Tumbleweed
2. Build Ghostty from source with the suggested command, `zig build -p $HOME/.local -Doptimize=ReleaseFast`.

**Comment by mitchellh** [comment]
Converted to Q&A. This is almost certainly a GTK issue (whether a bug or just configuration issue). Ghostty doesn't draw these elements, it just uses native GTK widgets. If there is something we discover we can do to mitigate this we can convert that specifically to an issue.

**Comment by kpetremann** [comment]
I encountered the same issue, on Tumbleweed too.

If it can help someone, I got it fixed by forcing antialiasing in fontconfig.

* Add this in `~/.config/fontconfig/fonts.conf`:
```
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
<fontconfig>
    <match target="font">
        <edit name="scalable" mode="assign">
            <bool>true</bool>
        </edit>
    </match>
</fontconfig>
```
* `fc-cache -f` might be needed.

note: I tried many changes in `~/.config/gtk-4.0/settings.ini` but it did not fix this particular issue.

**Comment by 00-kat** [comment]
This looks like messed up font configuration, or GTK messing up with specific fonts, or something.
